### ISSUE TITLE
How to test latency performance with different data size?

### ISSUE THREAD

**Issue by chaoli2** [ApexAI/performance_test#74]
I notice that there are multiple [message types](https://github.com/ApexAI/performance_test/tree/master/performance_test/src/msg) defined with different data size. I've tried "Array1k"(default size in README), "Array4k", "Array16k", "Array60k"(why noy 64k...), "Array1m" and "Array2m" with below command or similar:
```
ros2 run performance_test perf_test -c ROS2 -l log -t Array4k --max_runtime 120
```
Above message types all work well.

But when I move forward to "Array4m", I got below error message:
```
ros2 run performance_test perf_test -c ROS2 -l log -t Array4m --max_runtime 120

terminate called after throwing an instance of 'std::runtime_error'
  what(): a topic with the requested name does not exist or communication mean not supported.
```

When I switch my test to PointCloud message type, there are not any responding when executing below command
```
ros2 run performance_test perf_test -c ROS2 -l log -t PointCloud1m --max_runtime 120
```

What command and parameter should I use when testing PointCloud message type?

**Comment by deeplearningrobotics** [comment]
@chaoli2: I think there are two seperate problems here:

1. 
```
ros2 run performance_test perf_test -c ROS2 -l log -t Array4m --max_runtime 120

terminate called after throwing an instance of 'std::runtime_error'
  what(): a topic with the requested name does not exist or communication mean not supported.
```

This looks like a bug. Let me investigate.

2. Try to reduce the rate you send messages. Maybe your system is overloaded.

**Comment by chaoli2** [comment]
@deeplearningrobotics 
Thanks for your reply. How can I reduce the rate? Is there any parameter to do this?

**Comment by deeplearningrobotics** [comment]
@chaoli2: Yes. Running `ros2 run performance_test perf_test --help` will give you all the parameters. You should try `--rate 10` as a start which will publish 10 samples a second. The default I think is a 1000 samples.

**Comment by chaoli2** [comment]
@deeplearningrobotics 
Thanks. It works.
When I run
```
ros2 run performance_test perf_test -c ROS2 -l log -t PointCloud1m --max_runtime 120 --rate 10
```
I got an error:
```
terminate called after throwing an instance of 'rclcpp::exceptions::RCLError'
  what(): could not create publisher: Liveliness QoS is not yet supported for fastrtps., at /tmp/binarydeb/rosdashing-rmw-fastrtps-shared-cpp-0.7.4/src/qos.cpp:167, at /tmp/binarydeb/ros-dashing-rcl-0.7.5/src/rcl/publisher.c:171
```
Do you get any idea about this?

**Comment by dejanpan** [comment]
@chaoli2 the error is what it indeed is - FastRTPS in Dashing namely indeed does not have Liveliness QoS supported. 

@esteve @MiguelCompany do you know why is performance_test trying to use it?

**Comment by MiguelCompany** [comment]
> @esteve @MiguelCompany do you know why is performance_test trying to use it?

I think this is the same as in #66. Please see [my comment](https://github.com/ApexAI/performance_test/issues/66#issuecomment-514570691) there for a possible solution.

**Comment by chaoli2** [comment]
@MiguelCompany @dejanpan 
Thanks for your update. I got the PointCloud performance test results of "512K", "1m", "2m" and "4m", but failed when testing "PointCloud8m". 
```
ros2 run performance_test perf_test -c ROS2 -l log -t PointCloud8m --max_runtime 120 --rate 10

terminate called after throwing an instance of 'std::runtime_error'
  what(): a topic with the requested name does not exist or communication mean not supported.
```
